apps/explorer/src/ui/CopyToClipboard.tsx

```diff
@@ -28,31 +34,65 @@ const iconStyles = cva(['cursor-pointer hover:text-steel-dark'], {
 
 export type IconStylesProps = VariantProps<typeof iconStyles>;
 
-export interface CopyToClipboardProps extends IconStylesProps {
+export interface CopyToClipboardProps extends Omit<IconStylesProps, 'success'> {
 	copyText: string;
 	onSuccessMessage?: string;
 }
 
-const ICON_SIZES = {
+const COPY_ICON_SIZES = {
 	sm: Copy12,
 	md: Copy16,
 	lg: Copy24,
 };
 
+const CHECK_ICON_SIZES = {
+	sm: Check12,
+	md: CheckStroke16,
+	lg: CheckStroke24,
+};
+
+const TIMEOUT_TIMER = 2000;
+
 export function CopyToClipboard({
 	copyText,
 	color,
 	size = 'md',
 	onSuccessMessage = 'Copied!',
 }: CopyToClipboardProps) {
+	const [showCopyIcon, setShowCopyIcon] = useState(true);
 	const copyToClipBoard = useCopyToClipboard(() => toast.success(onSuccessMessage));
 
-	const CopyIcon = ICON_SIZES[size!];
+	const CopyIcon = COPY_ICON_SIZES[size!];
+	const CheckIcon = CHECK_ICON_SIZES[size!];
+
+	const onClickHandler = async () => {
+		await copyToClipBoard(copyText);
+		setShowCopyIcon(false);
+	};
+
+	useEffect(() => {
+		if (!showCopyIcon) {
+			const timeout = setTimeout(() => {
+				setShowCopyIcon(true);
```

Comment:
this feels like it should happen in the `onClickHandler` since it is triggered as a direct action of clicking the copy button as opposed to being a side effect of rendering. it would be nice to see if we could remove this effect :)

Reply:
it's being a setTimeout, we would need to clean it up with a `useEffect` block. So I figure either do a `useRef` and setTimeout inside the onClick, then clear it in the `useEffect` or just do everything there. No strong pref though.

Reply:
updated to useRef solution